src/pipelinedefinition.cpp

```diff
@@ -494,8 +494,8 @@ class NodeValidator {
         if (demultiplicatorNodeInfo.demultiplyCount.value() != 0) {
             // TODO -1 vs 0 as demultiplexing value
             if (shape[0] != 0) {
-                // 0 means that node accepts dynamic shape
-                if (!shape[0].match(demultiplicatorNodeInfo.demultiplyCount.value())) {
+                auto demultiplyDimension = demultiplicatorNodeInfo.demultiplyCount.value() ? Dimension(demultiplicatorNodeInfo.demultiplyCount.value()) : Dimension::any();
```

Comment:
`demultiplicatorNodeInfo.demultiplyCount.value()` cant be 0 anyway since we check it in L494